llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -3943,7 +3943,7 @@ LegalizerHelper::lower(MachineInstr &MI, unsigned TypeIdx, LLT LowerHintTy) {
     // target can override this with custom lowering and calling the
     // implementation functions.
     LLT Ty = MRI.getType(MI.getOperand(0).getReg());
-    if (LI.isLegalOrCustom({G_UMIN, Ty}))
+    if (LI.isLegalOrCustom({G_UMIN, Ty}) && LI.isLegalOrCustom({G_UMAX, Ty}))
```

Comment:
No, the unsigned expansion only uses umin not umax. Check out the comments in `lowerAddSubSatToMinMax`:
```
    // uadd.sat(a, b) -> a + umin(~a, b)
    // usub.sat(a, b) -> a - umin(a, b)
```
Please revert.

Reply:
I already did. Thanks!